File: src/Entity/Reclamation.php
<?php

namespace App\Entity;

use DateTime;
use DateTimeInterface;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;

/**
 * Reclamation
 *
 * @ORM\Table(name="reclamation", indexes={@ORM\Index(name="id_user", columns={"id_user"}), @ORM\Index(name="id_cat", columns={"id_cat"})})
 * @ORM\Entity
 */
class Reclamation
{
    /**
     * @var int
     *
     * @ORM\Column(name="id_rec", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idRec;

    /**
     * @var string
     *
     * @ORM\Column(name="username", type="string", length=50, nullable=false)
     */
    private $username;

    /**
     * @var string|null
     *
     * @ORM\Column(name="obj_rec", type="string", length=50, nullable=true)
     * @Assert\NotBlank(message="This field is obligatory.")
     * @Assert\Type(
     *     type="alpha",
     *     message="Please fill it with letters only."
     * )
     */
    private $objRec;

    /**
     * @var string|null
     *
     * @ORM\Column(name="suj_rec", type="string", length=150, nullable=true)
     * @Assert\NotBlank(message="This field is obligatory.")
     *
     */
    private $sujRec;

    /**
     * @var string|null
     *
     * @ORM\Column(name="etat_rec", type="string", length=20, nullable=true, options={"default"="To do"})
     */
    private $etatRec = 'To do';

    /**
     * @var DateTime|null
     *
     * @ORM\Column(name="date_rec", type="datetime", nullable=true, options={"default"="CURRENT_TIMESTAMP"})
     */
    private $dateRec ;


    /**
     * @var \App\Entity\Categories
     *
     * @ORM\ManyToOne(targetEntity=Categories::class, inversedBy="reclamations")
     * @ORM\JoinColumns({
     * @ORM\JoinColumn(name="id_cat", referencedColumnName="id_cat")
     * })
     * @Assert\NotNull(message="This field is obligatory.")
     */
    private $idCat;

    /**
     * @return int
     */
    public function getIdRec(): int
    {
        return $this->idRec;
    }

    /**
     * @param int $idRec
     */
    public function setIdRec(int $idRec): void
    {
        $this->idRec = $idRec;
    }

    /**
     * @return string
     */
    public function getUsername(): ?string
    {
        return $this->username;
    }

    /**
     * @param string $username
     */
    public function setUsername(string $username): void
    {
        $this->username = $username;
    }

    /**
     * @return string|null
     */
    public function getObjRec(): ?string
    {
        return $this->objRec;
    }

    /**
     * @param string|null $objRec
     */
    public function setObjRec(?string $objRec): void
    {
        $this->objRec = $objRec;
    }

    /**
     * @return string|null
     */
    public function getSujRec(): ?string
    {
        return $this->sujRec;
    }

    /**
     * @param string|null $sujRec
     */
    public function setSujRec(?string $sujRec): void
    {
        $this->sujRec = $sujRec;
    }

    /**
     * @return string|null
     */
    public function getEtatRec(): ?string
    {
        return $this->etatRec;
    }

    /**
     * @param string|null $etatRec
     */
    public function setEtatRec(?string $etatRec): void
    {
        $this->etatRec = $etatRec;
    }

    /**
     * @return DateTimeInterface|null
     */
    public function getDateRec(): ?DateTimeInterface
    {
        return $this->dateRec;
    }

    /**
     * @param DateTimeInterface|null $dateRec
     */
    public function setDateRec(?DateTimeInterface $dateRec): void
    {
        $this->dateRec = $dateRec;
    }

    /**
     * @return \App\Entity\Categories
     */
    public function getIdCat(): ?\App\Entity\Categories
    {
        return $this->idCat;
    }

    /**
     * @param \App\Entity\Categories $idCat
     */
    public function setIdCat(\App\Entity\Categories $idCat): void
    {
        $this->idCat = $idCat;
    }

    public function __toString()
    {
        return $this->username;
    }


}

File: src/entities/Reclamation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Date;

/**
 *
 * @author salma
 */
public class Reclamation {
    private int id_rec;
    private String username;
    private String obj_rec;
    private String suj_rec;
    private String etat_rec;
    private int id_cat;
    private String nom_cat;
    private String date_rec;
   

   

    public Reclamation(int id_rec, int id_cat, String nom_cat, String username, String obj_rec, String suj_rec, String etat_rec, String date_rec) {
        this.id_rec = id_rec;
        this.id_cat = id_cat;
        this.nom_cat = nom_cat;
        this.username = username;
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
        this.etat_rec = etat_rec;
        this.date_rec = date_rec;
    }

    public Reclamation() {
    }

    public Reclamation(int id_rec) {
        this.id_rec = id_rec;
    }

    public Reclamation(int id_cat, String nom_cat, String username, String obj_rec, String suj_rec) {
        this.id_cat = id_cat;
        this.nom_cat = nom_cat;
        this.username = username;
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
    }
    
    public Reclamation(String nom_cat, String username, String obj_rec, String suj_rec) {
        this.nom_cat = nom_cat;
        this.username = username;
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
    }

    public Reclamation(int id_rec, String etat_rec) {
        this.id_rec = id_rec;
        this.etat_rec = etat_rec;
    }
    
    public Reclamation(String obj_rec, String suj_rec, int id_cat) {
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
        this.id_cat=id_cat;
    }
    
    public Reclamation(String obj_rec, String suj_rec) {
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
        
    }

    public Reclamation(int id_rec, String nom_cat, String username, String obj_rec, String suj_rec, String etat_rec, String date_rec) {
        this.id_rec = id_rec;
        this.nom_cat = nom_cat;
        this.username = username;
        this.obj_rec = obj_rec;
        this.suj_rec = suj_rec;
        this.etat_rec = etat_rec;
        this.date_rec = date_rec;
    }

//    public Reclamation(String obj_rec, String suj_rec, String nom_cat) {
//        this.obj_rec = obj_rec;
//        this.nom_cat = nom_cat;
//        this.suj_rec = suj_rec;
//    }

    public int getId_rec() {
        return id_rec;
    }

    public void setId_rec(int id_rec) {
        this.id_rec = id_rec;
    }

    public int getId_cat() {
        return id_cat;
    }

    public void setId_cat(int id_cat) {
        this.id_cat = id_cat;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getObj_rec() {
        return obj_rec;
    }

    public void setObj_rec(String obj_rec) {
        this.obj_rec = obj_rec;
    }

    public String getSuj_rec() {
        return suj_rec;
    }

    public void setSuj_rec(String suj_rec) {
        this.suj_rec = suj_rec;
    }

    public String getEtat_rec() {
        return etat_rec;
    }

    public String getNom_cat() {
        return nom_cat;
    }

    public void setNom_cat(String nom_cat) {
        this.nom_cat = nom_cat;
    }

    public void setEtat_rec(String etat_rec) {
        this.etat_rec = etat_rec;
    }

    public String getDate_rec() {
        return date_rec;
    }

    public void setDate_rec(String date_rec) {
        this.date_rec = date_rec;
    }

    
    @Override
    public String toString() {
        return "Reclamation : " + "User : " + username + " Objet : " + obj_rec + " Description : " + suj_rec + " Etat : " + etat_rec + " Catégorie : " + id_cat + " Temps d'envoi : " + date_rec ;
    }

    

    
    
}